### DISCUSSION TITLE
How to make o2m (m2m) custom display

### DISCUSSION THREAD

**Discussion by IlyaPetrovM** [directus/directus#7457] [Archive]
Hello! I want to make my own display for o2m or m2m field. 
I'm new to Vue. And I don't understand, what I have to change in example from [docs](https://docs.directus.io/guides/displays/).
This is example from docs:

**src/display.vue**

```vue
<template>
	<div>Value: {{ value }}</div>
</template>

<script>
export default {
	props: {
		value: String,
	},
};
</script>
```
How can I get related values from another collection?
Should I set `value: Object `  ? 

I've noticed [there](https://github.com/directus/directus/blob/19424feefafe5a32f7f34f8c39b1d8da1ddf1828/app/src/displays/related-values/related-values.vue#L67), that you use "utils" lib to compute related values

How to import this module to my custom display project?

` import getRelatedCollection from '@/utils/get-related-collection'; `
is not working obviously

**Comment by rijkvanzanten** [comment]
You can use the `fields` handler in the index.ts file of the display to tell the layout what fields to fetch for that particular data point:

https://github.com/directus/directus/blob/dfb91bbb0b6e71fe6bac259d905c1bf51d0295fe/app/src/displays/related-values/index.ts#L23-L38

```
fields() {
  return ['id', 'title']
}
```

will tell the layout to fetch the nested id and title fields.

The `getRelatedCollection` utility function is not currently exported to extensions (cc @nickrum)
